src/librbd/ImageCtx.cc

```diff
@@ -801,13 +802,17 @@ class SafeTimerSingleton : public SafeTimer {
     ASSIGN_OPTION(mtime_update_interval, uint64_t);
     ASSIGN_OPTION(atime_update_interval, uint64_t);
     ASSIGN_OPTION(skip_partial_discard, bool);
+    ASSIGN_OPTION(discard_granularity_bytes, uint64_t);
     ASSIGN_OPTION(blkin_trace_all, bool);
 
 #undef ASSIGN_OPTION
 
     if (sparse_read_threshold_bytes == 0) {
       sparse_read_threshold_bytes = get_object_size();
     }
+    if (!skip_partial_discard) {
+      discard_granularity_bytes = 0;
```

Comment:
I started to allow that as well but since this is an internal-only value (externally enforced to be 4KiB<=x<=32MiB), I felt it was OK (and protects against accidental division by zero issues)